### DISCUSSION TITLE
pip_requirements not picking the correct datasource url if requirements.txt is inside the folder

### DISCUSSION THREAD

**Discussion by Rahul-learn** [renovatebot/renovate#11933] [Request Help]
**Env:** openshift
**renovate version:** 24.32.2
**Host:** Self hosted

If we keep requirements.txt at root level  renovate is working fine but when we move requirements.txt inside the folder, its not picking the correct datasource URL and its not creating any PR.

**Comment by viceice** [comment]
V24 of renovate is very outdated and no longer supported. please upgrade to a recent v27.

**Comment by Rahul-learn** [comment]
@viceice 
updated version to v27 still we are seeing the same issue

**Reply by viceice** [reply]
Can you provide a some reproduction, as we need more information. Debug log messages would maybe help.

**Reply by Rahul-learn** [reply]
I am seeing only below statement in logs

INFO: Repository started (repository=*****/****)
       "renovateVersion": "27.19.0"
 INFO: Dependency extraction complete (repository=*****/*****)
       "baseBranch": "****",
       "stats": {
         "managers": {"pip_requirements": {"fileCount": 1, "depCount": 11}},
         "total": {"fileCount": 1, "depCount": 11}
       }
 INFO: Repository finished (repository=*****/*****)
       "durationMs": 4107

**Reply by viceice** [reply]
Please run with `LOG_LEVEL=debug` in env to get debug logs

**Reply by Rahul-learn** [reply]
I have enabled debug logs. This is my observation If I moved requirements.txt inside the folder in the registry URL extra space is getting added as highlighted below. Same requirement.txt is working if I keep in root level.

 https://nexus.******.com/repository/pypi/simple/**%0D**/Click/

**Reply by Rahul-learn** [reply]
@viceice 

Do I need to change anything in the configuration?

**Reply by viceice** [reply]
Please provide your sanitized configs and a small sample requirements.txt file

**Reply by Rahul-learn** [reply]
@viceice 
Please find the config & sample requirements.txt

**ConfigMap**


  config.json:  { "platform": "github", "endpoint": "https://github.******.com/api/v3",
    "LOG_LEVEL": "debug", "includeForks": true, "persistRepoData": true, "enabledManagers": [ "gradle", "gradle-wrapper", "npm", "pip_requirements", "pip_setup", "pipenv", "poetry"], "packageRules": 
    [ { "datasources": [ "python" ], "registryUrls": [ "https://nexus.*******.com/repository/pypi/simple/"
    ] } ], "repositories": [ "*****/****" ] }

**requirements.txt**

setuptools==42.0.1
certifi==2019.3.9
chardet==3.0.4
Click==7.0

**Reply by Rahul-learn** [reply]
@viceice 

any idea why its not picking the correct datasource url?
